modules/otel-collector/main.tf

```diff
@@ -0,0 +1,225 @@
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+
+terraform {
+  required_version = ">=1.0.0"
+
+  required_providers {
+    helm = {
+      source  = "hashicorp/helm"
+      version = ">=2.2.0"
+    }
+  }
+}
+
+
+
+### Module defaults are managed below:
+locals {
+  aws_relay_conf = {
+    exporters = {
+      awsxray = {
+        index_all_attributes = true
+      }
+    }
+    extensions = {
+      health_check = {}
+    }
+    processors = {
+      batch = {}
+      memory_limiter = {
+        ballast_size_mib = 100
+        check_interval   = "5s"
+        limit_mib        = 200
+        spike_limit_mib  = 62
+      }
+    }
+    receivers = {
+      otlp = {
+        protocols = {
+          grpc = null
+          http = null
+        }
+      }
+    }
+    service = {
+      extensions = [
+        "health_check"
+      ]
+      pipelines = {
+        traces = {
+          exporters = [
+            "awsxray"
+          ]
+          processors = [
+            "memory_limiter",
+            "batch"
+          ],
+          receivers = [
+            "otlp"
+          ]
+        }
+      }
+    }
+  }
+
+  gcp_relay_conf = {
+    exporters = {
+      googlecloud = {}
+    }
+    extensions = {
+      health_check = {}
+    }
+    processors = {
+      batch = {}
+      memory_limiter = {
+        ballast_size_mib = 100
+        check_interval   = "5s"
+        limit_mib        = 200
+        spike_limit_mib  = 62
+      }
+    }
+    receivers = {
+      otlp = {
+        protocols = {
+          grpc = null
+          http = null
+        }
+      }
+    }
+    service = {
+      extensions = [
+        "health_check"
+      ]
+      pipelines = {
+        traces = {
+          receivers = [
+            "otlp"
+          ]
+          processors = [
+            "memory_limiter",
+            "batch"
+          ]
+          exporters = [
+            "googlecloud"
+          ]
+        }
+        metrics = {
+          receivers = [
+            "otlp"
+          ]
+          processors = [
+            "memory_limiter",
+            "batch"
+          ]
+          exporters = [
+            "googlecloud"
+          ]
+        }
+      }
+    }
+  }
+
+  relay_config = var.cloud_provider == "aws" ? local.aws_relay_conf : (var.cloud_provider == "gcp" ? local.gcp_relay_conf : null)
```

Comment:
I don't understand this "relay" stuff.  I see "exporters" ([ref](https://github.com/open-telemetry/opentelemetry-helm-charts/blob/opentelemetry-operator-0.4.0/charts/opentelemetry-collector/values.yaml#L9)).

Seems to me that the receivers (zipkin, etc) will be the same in all deployments, and the exporters will vary by cloud.  

Hopefully we can drop the `otel_relay_configmap`.